pkg/kn/commands/source/apiserver/create_flag.go

```diff
@@ -0,0 +1,73 @@
+// Copyright © 2019 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apiserver
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/spf13/cobra"
+	sources_v1alpha1 "knative.dev/eventing/pkg/apis/sources/v1alpha1"
+)
+
+const ApiVersionSplitStr = ":"
+const ApiServerSourceArraySplitStr = ","
+
+// ApiServerSourceUpdateFlags are flags for create and update a ApiServerSource
+type ApiServerSourceUpdateFlags struct {
+	ServiceAccountName string
+	Mode               string
+	ApiVersion         string
+	Resources          string
+}
+
+// GetApiServerResourceArray is to return an array of ApiServerResource from a string. A sample is Event:v1:true,Pod:v2:false
+func (f *ApiServerSourceUpdateFlags) GetApiServerResourceArray() []sources_v1alpha1.ApiServerResource {
+	var resourceList []sources_v1alpha1.ApiServerResource
+	for _, r := range strings.Split(f.Resources, ApiServerSourceArraySplitStr) {
+		version, kind, controller := getValidResource(r)
+		resourceRef := sources_v1alpha1.ApiServerResource{
+			APIVersion: version,
+			Kind:       kind,
+			Controller: controller,
+		}
+		resourceList = append(resourceList, resourceRef)
+	}
+	return resourceList
+}
+
+func getValidResource(resource string) (string, string, bool) {
+	var version, kind string
+	var isController bool
+	var err error
+	parts := strings.Split(resource, ApiVersionSplitStr)
+	if len(parts) == 3 {
+		kind = parts[0]
+		version = parts[1]
+		isController, err = strconv.ParseBool(parts[2])
+		if err != nil {
+			isController = false
+		}
+	}
+	return version, kind, isController
+}
+
+//AddApiServerFlags is to set parameters
+func (f *ApiServerSourceUpdateFlags) Add(cmd *cobra.Command) {
+	cmd.Flags().StringVar(&f.ServiceAccountName, "service-account", "", "Name of the ServiceAccount to use to run this source")
+	cmd.Flags().StringVar(&f.Mode, "mode", "Ref", "'Ref' or 'Resource', the receive adapter controller runs under.")
+	cmd.Flags().StringVar(&f.ApiVersion, "apiVersion", "1.0", "Kubernetes API version of the resource to watch")
+	cmd.Flags().StringVar(&f.Resources, "resources", "r", "comma seperate Kind:APIVersion:isController list, e.g. Event:v1:true")
```

Comment:
should add an example like documented in code `Event:v1:true,Pod:v2:false`

Reply:
done